config/checker_profile_map.json

```diff
@@ -0,0 +1,271 @@
+{
+  "available_profiles": {
```

Comment:
Also it would be great to extend the command line help message with profiles and guidelines here: https://github.com/Ericsson/codechecker/blob/0a7617b897f55827a162ab76cecfba2a370a3e76/analyzer/codechecker_analyzer/cmd/analyze.py#L604
Maybe we can create a separate section here and give a short description about it:
```
Profiles
------------------------------------------------
....

Guidelines
------------------------------------------------
....
```